force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,77 @@
+public with sharing class OpportunityTriggerHandler extends TriggerHandler {
+    private List<Opportunity> newOppList;
+    private List<Opportunity> oldOppList;
+    private Map<Id,Opportunity> newOppMap;
+    private Map<Id,Opportunity> oldOppMap;
+    
+    public OpportunityTriggerHandler() {
+        this.newOppList = (List<Opportunity>)Trigger.new;
+        this.oldOppList = (List<Opportunity>)Trigger.old;
+        this.newOppMap = (Map<Id,Opportunity>)Trigger.newMap;
+        this.oldOppMap = (Map<Id,Opportunity>)Trigger.oldMap;
+
+    }
+    
+    public override void beforeUpdate(){
+        opportunityAmountValidation(newOppList);
+        setOpportunityPrimaryContact(newOppList);
+    }
+
+    public override void beforeDelete(){
+        deleteClosedWonBankingOpportunityValidation(oldOppList);
+    }
+
+
+    public static void opportunityAmountValidation(List<Opportunity> oppList){
+        for(Opportunity eachOpp :oppList){
+            if(eachOpp.Amount < 5000){
```

Comment:
Try putting the number 5000 into a variable so it's not a magic number. Same can be done with the error text.